src/Storefront/Controller/CartLineItemController.php

```diff
@@ -69,6 +73,24 @@ public function deleteLineItem(Cart $cart, string $id, Request $request, SalesCh
         });
     }
 
+    #[Route(path: '/checkout/line-item/delete', name: 'frontend.checkout.line-item.delete', defaults: ['XmlHttpRequest' => true], methods: ['POST', 'DELETE'])]
+    public function deleteLineItems(Cart $cart, Request $request, SalesChannelContext $context): Response
+    {
+        return Profiler::trace('cart::delete-line-items', function () use ($cart, $request, $context) {
+            try {
+                $this->cartItemRemoveRoute->remove($request, $cart, $context);
```

Comment:
Don't pass the request directly to the store-api. Prepare the ids as an array and pass them to the cartService to a new method removeItems
```php
/**
     * @throws CartException
     * @param array<int, string> $identifier
     */
    public function removeItems(Cart $cart, array $identifiers, SalesChannelContext $context): Cart
    {
        $request = new Request();
        $request->request->set('ids', $identifiers);

        $cart = $this->itemRemoveRoute->remove($request, $cart, $context)->getCart();

        return $this->cart[$cart->getToken()] = $cart;
    }
```

Reply:
I extended the existing `remove` method. Hope it is okay in this way, it is the same pattern as for the `add` method.

Reply:
As the `CartService` is not `final`, it would be a BC break... Hence I added the new method as suggested and internally use it also in the `remove` method. 
Not sure if the `removeItems` should have the Union Type or not, and/or if the `remove` method should be deprecated.